### PR TITLE
K8s moved its registry

### LINKED ISSUE #32041: K8s is moving it's images
### Describe the bug

As described in this [blog post](https://kubernetes.io/blog/2023/03/10/image-registry-redirect/) kubernetes is moving it's registry to eventually sunset the old one.

When starting a devservice for k8s we should use the image from the new registry

### Expected behavior

Dev service starts container
registry.k8s.io/kube-apiserver

### Actual behavior

Dev service starts container
k8s.gcr.io/kube-apiserver

### How to Reproduce?

Build an app with extension quarkus-kubernetes-client and run the dev mode

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by bulldog98**
old registry k8s.grc.io will be phased out use registry.k8s.io instead

fixes #32041

### CODE REVIEW

File: extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/DevServicesKubernetesProcessor.java [hunk 1]
```diff
@@ -343,7 +343,7 @@ public Map<String, String> getKubeconfig() {
                 return getKubernetesClientConfigFromKubeConfig(
                         KubeConfigUtils.parseKubeConfig(KubeConfigUtils.replaceServerInKubeconfig(containerAddress.getUrl(),
                                 getFileContentFromContainer(KIND_KUBECONFIG))));
-            } else if (image.contains("k8s.gcr.io/kube-apiserver")) {
+            } else if (image.contains("registry.k8s.io/kube-apiserver")) {
```
Comment (gsmet) [inline extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/DevServicesKubernetesProcessor.java:346]: Shouldn't we support both for a while?

Comment (bulldog98) [reply extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/DevServicesKubernetesProcessor.java:346]: did not think about that, I'll have a look tomorrow, to tired right now.

Comment (gsmet) [reply extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/DevServicesKubernetesProcessor.java]: I adjusted it and force pushed!

**Review by gsmet — APPROVED** [review]
Thanks, let's get this in once CI is green.

**Pull request merged**